File: component/biz/Class_biz_role_privilege_map.pas
unit Class_biz_role_privilege_map;

interface

uses
  Class_db_role_privilege_map,
  Class_biz_notifications,
  system.collections;

type
  TClass_biz_role_privilege_map = class
  private
    db_role_privilege_map: TClass_db_role_privilege_map;
    biz_notifications: TClass_biz_notifications;
  public
    constructor Create;
    procedure Bind
      (
      tier_filter: string;
      sort_order: string;
      be_sort_order_descending: boolean;
      target: system.object;
      out crosstab_metadata: arraylist
      );
    procedure Save
      (
      member_id: string;
      role_id: string;
      be_granted: boolean
      );
  end;

implementation

uses
  kix;

constructor TClass_biz_role_privilege_map.Create;
begin
  inherited Create;
  db_role_privilege_map := TClass_db_role_privilege_map.Create;
  biz_notifications := TClass_biz_notifications.Create;
end;

procedure TClass_biz_role_privilege_map.Bind
  (
  tier_filter: string;
  sort_order: string;
  be_sort_order_descending: boolean;
  target: system.object;
  out crosstab_metadata: arraylist
  );
var
  tier_quoted_value_list: string;
begin
  if tier_filter = '2' then begin
    tier_quoted_value_list := '2' + QUOTE + COMMA + QUOTE + '3';
  end else begin
    tier_quoted_value_list := tier_filter;
  end;
  tier_quoted_value_list := QUOTE + tier_quoted_value_list + QUOTE;
  db_role_privilege_map.Bind(tier_quoted_value_list,sort_order,be_sort_order_descending,target,crosstab_metadata);
end;

procedure TClass_biz_role_privilege_map.Save
  (
  member_id: string;
  role_id: string;
  be_granted: boolean
  );
begin
  db_role_privilege_map.Save(member_id,role_id,be_granted);
end;

end.

File: component/biz/Class_biz_role_privilege_map.cs
using Class_biz_notifications;
using Class_biz_roles;
using Class_db_role_privilege_map;
using kix;
using System;
using System.Collections;

namespace Class_biz_role_privilege_map
{
    public class TClass_biz_role_privilege_map
    {
        private TClass_db_role_privilege_map db_role_privilege_map = null;
        private TClass_biz_notifications biz_notifications = null;
        private TClass_biz_roles biz_roles = null;

        //Constructor  Create()
        public TClass_biz_role_privilege_map() : base()
        {
            db_role_privilege_map = new TClass_db_role_privilege_map();
            biz_notifications = new TClass_biz_notifications();
            biz_roles = new TClass_biz_roles();
        }

        internal bool BeOkToDefaultTierFilterToDepartment
          (
          bool be_ok_to_see_all_squads,
          string[] roles
          )
          {
          return be_ok_to_see_all_squads && (int.Parse(biz_roles.TierOfName(roles[0])) == 1);
          }

        public void Bind(string tier_filter, string sort_order, bool be_sort_order_descending, object target, out ArrayList crosstab_metadata)
        {
            string tier_quoted_value_list;
            if (tier_filter == "2")
            {
                tier_quoted_value_list = "2" + k.QUOTE + k.COMMA + k.QUOTE + "3";
            }
            else
            {
                tier_quoted_value_list = tier_filter;
            }
            tier_quoted_value_list = k.QUOTE + tier_quoted_value_list + k.QUOTE;
            db_role_privilege_map.Bind(tier_quoted_value_list, sort_order, be_sort_order_descending, target, out crosstab_metadata);
        }

        public void BindActuals(string tier_filter, string sort_order, bool be_sort_order_ascending, object target)
        {
            string tier_quoted_value_list;
            if (tier_filter == "2")
            {
                tier_quoted_value_list = "2" + k.QUOTE + k.COMMA + k.QUOTE + "3";
            }
            else
            {
                tier_quoted_value_list = tier_filter;
            }
            tier_quoted_value_list = k.QUOTE + tier_quoted_value_list + k.QUOTE;
            db_role_privilege_map.BindActuals(tier_quoted_value_list, sort_order, be_sort_order_ascending, target);
        }

        public void Save(string member_id, string role_id, bool be_granted)
        {
            db_role_privilege_map.Save(member_id, role_id, be_granted);
        }

    } // end TClass_biz_role_privilege_map

}
